### PR TITLE
Add table to display engine search, suggest, and trending URLs.

### PR DESCRIPTION
**Pull request by mandysGit**
Method and results not added in the tablet yet. We discussed adding those in another patch. 

**Light Mode:**
Screenshot 2024-06-28 at 3 54 42 PM

**Dark Mode:** 
Screenshot 2024-06-28 at 3 54 11 PM

### CODE REVIEW

**Review by Standard8 — CHANGES_REQUESTED** [review]
Thank you, this is looking good and definitely will be nice to have. I've commented in-line for a few places where I think we can improve it further.

File: extension/content/engineUrlView.mjs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  constructor() {
+    super();
+  }
+
+  async loadEngineUrls(config) {
+    const COL_HEADERS = [
+      "URL Type",
+      `Full URL for ${config.name} `,
+      "Method",
+      "Results",
```
[the remark this answers is not in the document]
Comment (mandysGit) [reply extension/content/engineUrlView.mjs:15]: 👍 

I'll create a follow up bug for this.

[the remark this answers is not in the document]
Comment (mandysGit) [reply extension/content/engineUrlView.mjs:15]: [Bug 1909017 - Update search engine dev tools to show results from suggest or trending urls](https://bugzilla.mozilla.org/show_bug.cgi?id=1909017)

File: extension/content/engineUrlView.mjs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    this.appendChild(fragment);
+
+    return fragment;
+  }
+
+  createTableFragment(headers, rowHeaders, sortedUrls) {
+    let fragment = document.createDocumentFragment();
+    let table = document.createElement("table");
+    let thead = document.createElement("thead");
+    let tbody = this.createTableBody(rowHeaders);
```
[the remark this answers is not in the document]
Comment (mandysGit) [reply extension/content/engineUrlView.mjs:37]: I used `createThead`, `createTBody`, `insertRow` methods where I could.

For some of the elements, I still needed to use `document.createElement`.

**Comment by mandysGit** [comment]
Addressed in-line review comments. This is ready for another review.

**Review by Standard8 — COMMENTED** [review]
I added the extra tweaks I suggested to clear the table when other items are suggested. We should probably fix the flickering at some stage, but this is a good improvement to get landed.

**Review by Standard8 — APPROVED** [review]

**Pull request merged**